### ISSUE TITLE
Unable to create index with qualified index name on Oracle dialect

### ISSUE THREAD

**Issue by nielsmeijer** [jOOQ/jOOQ#9414]
I stumbled on a regression while creating indexes with a qualified name using the Oracle jOOQ dialect. In jOOQ ```3.11.11 (pro)```, the qualified name is rendered correctly in Oracle. After upgrading, indexes are no longer using the qualified name. 
This is a problem when using the same index name in different database schemas ( ```ORA-00955: name is already used by an existing object```). 
When the index is created with the unqualified name,  the index is placed in the connecting users schema. 

It seems to introduced with or related to jOOQ issue #8723

## Code sample:

```java
DSL.using(SQLDialect.ORACLE12C)
.createIndex(name("schema_name", "index_name"))
.on(table, field)
.execute();
```

When using jOOQ ```3.11.11 (pro)```, above code sample translates to 

```sql
create index "schema_name"."index_name" on "schema_name"."table_name"("field_name")
```

After upgrading to jOOQ ```3.12.1 (pro)```, the same code sample translates to 

```sql
create index "index_name" on "schema_name"."table_name"("field_name")
``` 

### Versions:

- jOOQ: tested on ```3.11.11 (pro)``` and ```3.12.1 (pro)```
- Database: ```Oracle Database 12c Enterprise Edition Release 12.2.0.1.0 ```
- JDBC Driver: oracle ojdbc7

**Comment by lukaseder** [comment]
Thanks a lot for your report. We'll have a look at this ASAP early next week. We're planning a 3.12.3 release soon as well.

**Comment by lukaseder** [comment]
We didn't have an integration test for multi schema indexes yet, which is why this regression went undetected. Probably a change was made based on the wrong assumption that an index is always located with the table in the same schema. But of course, this does not take into account that a different user might be creating the index than the owner of the table.

**Comment by lukaseder** [comment]
Related bug discovery: #9454

**Comment by lukaseder** [comment]
The regression seems to have been introduced with https://github.com/jOOQ/jOOQ/issues/8723

**Comment by lukaseder** [comment]
Hmm, there really isn't a good reason not to render the qualified index name in Oracle, if it is available:
https://github.com/jOOQ/jOOQ/blob/41973fa392f2c28dfe806930dfb9901072fd9414/jOOQ/src/main/java/org/jooq/impl/IndexImpl.java#L104-L111

The dialects that explicitly do not support index qualification in `create index` statements are already listed in the relevant flag:
https://github.com/jOOQ/jOOQ/blob/41973fa392f2c28dfe806930dfb9901072fd9414/jOOQ/src/main/java/org/jooq/impl/IndexImpl.java#L79

So, this should be an easy fix.

**Comment by lukaseder** [comment]
A workaround is to set the current schema of your session prior to creating an index. That might be a good idea in general, in case some explicit qualification is forgotten, or lost:

```sql
alter session set current_schema = "schema_name";
```

**Comment by lukaseder** [comment]
jOOQ supports this via

```java
DSL.using(connection, SQLDialect.ORACLE12C)
.setSchema(name("schema_name"))
.execute();
```

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.13.0. Backport scheduled for 3.12.3 (#9455, due this week)
